examples/Google.Cloud.Functions.Examples.TimeZoneConverter/TimeZoneConverter.cs

```diff
@@ -0,0 +1,157 @@
+
+            // Check that we have all the parameters we expect.
+            if (fromZoneParameter is null || toZoneParameter is null || whenParameter is null)
+            {
+                return null;
+            }
+
+            // Check that they all map to correct time zones and we can parse the "when" value as a LocalDateTime
+            return
+                _timeZoneProvider.GetZoneOrNull(fromZoneParameter) is DateTimeZone fromZone &&
+                _timeZoneProvider.GetZoneOrNull(toZoneParameter) is DateTimeZone toZone &&
+                LocalPattern.Parse(whenParameter) is { Success: true, Value: var when }
+                ? Convert(when, fromZone, toZone)
+                : null;
+        }
+
+        /// <summary>
+        /// Performs the actual conversion, with validated inputs. (A "valid local date/time that happens to be skipped
+        /// in the zone we're mapping from" still counts as skipped.)
+        /// </summary>
+        /// <param name="when">The parsed value from the "when" query parameter</param>
+        /// <param name="fromZone">The time zone corresponding to the ID in the "fromZone" query parameter</param>
+        /// <param name="toZone">The time zone corresponding to the ID in the "toZone" query parameter</param>
+        /// <returns>The result of the conversion</returns>
+        private ConversionResult Convert(LocalDateTime when, DateTimeZone fromZone, DateTimeZone toZone)
+        {
+            var mapping = fromZone.MapLocal(when);
+            return mapping.Count switch
+            {
+                0 => ConvertSkipped(),
+                1 => ConvertUnambiguous(),
+                2 => ConvertAmbiguousInput(),
+                _ => throw new InvalidOperationException($"{mapping.Count} mappings - bug in NodaTime?")
+            };
+
+            ConversionResult ConvertSkipped()
+            {
+                var shiftedForward = Resolvers.ReturnForwardShifted(when, fromZone, mapping.EarlyInterval, mapping.LateInterval);
+                var converted = shiftedForward.WithZone(toZone).LocalDateTime;
+                return CreateResult(converted, ConversionType.SkippedInputForwardShiftedResult);
+            }
+
+            ConversionResult ConvertUnambiguous()
+            {
+                var converted = mapping.First().WithZone(toZone).LocalDateTime;
+                return CreateResult(converted, ConversionType.Unambiguous);
+            }
+
+            ConversionResult ConvertAmbiguousInput()
+            {
+                var earlyConverted = mapping.First().WithZone(toZone).LocalDateTime;
+                var lateConverted = mapping.Last().WithZone(toZone).LocalDateTime;
+                return CreateResult(
+                    // While it would be very odd for the results of converting the earlier instant to be
+                    // later than the results of the converting the later instant, it's theoretically possible.
```

Comment:
nit: than the results of _(the)_ converting

Reply:
Done, thank.s